src/NJsonSchema/Validation/JsonSchemaValidatorOptions.cs

```diff
@@ -0,0 +1,11 @@
+﻿using System;
+
+namespace NJsonSchema.Validation
+{
+    /// <summary>Class to configure the behavior of <see cref="JsonSchemaValidator"/>. </summary>
+    public class JsonSchemaValidatorOptions
+    {
+        /// <summary>Whether to ignore casing for object properties.</summary>
+        public bool IgnorePropertyNameCase { get; set; }
```

Comment:
would it make sense to make this `public StringComparer PropertyNameComparer`?

Reply:
I don't mind either way. A `StringComparer` is obviously more flexible. @RicoSuter ?

Reply:
Fine for me either way.. as long it's an option.
To me it does not make much sense to allow case insensitive validation - it's probably against JSON Schema rules :-)

Reply:
StringComparer has the advantage that you can write a comparer like `FirstCharIgnoringComparer` which is convenient in some interop scenarios between JSON and C# classes - not sure if it applies here.

Reply:
I updated the PR with `StringComparer`

To me, I've always used case insensitive deserialization from JSON -> C# classes for convenience. Now I have a scenario where I want to use JsonSchema to validate certain JSON structures, and I want to preserve the case insensitivity for consistency